### PR TITLE
Use .rustfmt.toml

### LINKED ISSUE #2512: Use `.rustfmt.toml`
**Issue summary**

There're a few useful options in `.rustfmt.toml` whose default values are `preserve`, we could override them to get more consistent code styles.

Doc: https://rust-lang.github.io/rustfmt/

Options with `preserve` default values:
- [ ] [imports_granularity](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#imports_granularity)
- [ ] [group_imports](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#group_imports)
- [ ] [hex_literal_case](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#hex_literal_case)

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
## Summary of changes

Changes introduced in this pull request:
- 

## Reference issue to close (if applicable)

Closes #2512

## Change checklist

- [ ] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation,
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -1 +1,5 @@
 edition = "2021"
+
+imports_granularity="Module"
```
Comment (hanabi1224) [inline rustfmt.toml:3]: Please add the link to the doc in comments

Comment (hanabi1224) [inline rustfmt.toml:3]: Personally, I would prefer `imports_granularity="Crate"`

File: rustfmt.toml [hunk 2]
```diff
@@ -1 +1,5 @@
...
+group_imports="One"
```
Comment (hanabi1224) [inline rustfmt.toml:4]: Personally, I would prefer `imports_granularity="StdExternalCrate"`

File: rustfmt.toml [hunk 3]
```diff
@@ -1 +1,5 @@
 edition = "2021"
```
Comment (hanabi1224) [inline rustfmt.toml]: According to the doc, the default strategy of resolving `edition` looks good, why would we override it?

File: rustfmt.toml (hunk 2 above)
Comment (tyshko5) [reply rustfmt.toml]: Fixed

File: rustfmt.toml (hunk 1 above)
Comment (tyshko5) [reply rustfmt.toml]: Fixed

Comment (tyshko5) [reply rustfmt.toml]: Added

File: rustfmt.toml [hunk 4]
```diff
@@ -1 +1,5 @@
...
+
```
[the remark this answers is not in the document]
Comment (tyshko5) [reply rustfmt.toml]: renamed

**Review by hanabi1224 — APPROVED** [review]

File: .rustfmt.toml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Doc: https://rust-lang.github.io/rustfmt/
+imports_granularity="Crate"
+group_imports="StdExternalCrate"
+hex_literal_case="Lower"
```
Comment (hanabi1224) [inline .rustfmt.toml:4]: In additional to these, I think `wrap_comments = true` is also helpful

Comment (tyshko5) [reply .rustfmt.toml]: Added

**Review by LesnyRumcajs — COMMENTED** [review]
It's a bit weird that the max comment length is shorter than the max line length. Can we increase the former? Or do you think it's better this way?

**Comment by tyshko5** [comment]
> It's a bit weird that the max comment length is shorter than the max line length. Can we increase the former? Or do you think it's better this way?

I guess yes, we had too long comment lines @LesnyRumcajs

**Comment by hanabi1224** [comment]
> I guess yes, we had too long comment lines @LesnyRumcajs

@tyshko5 I think what @LesnyRumcajs means is the default [comment_width](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#comment_width) being shorter than [max_width](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#max_width)

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
> > I guess yes, we had too long comment lines @LesnyRumcajs
> 
> @tyshko5 I think what @LesnyRumcajs means is the default [comment_width](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#comment_width) being shorter than [max_width](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#max_width)

Yeah, exactly. Personally, I'd change it but it's not a huge deal.

**Comment by hanabi1224** [comment]
They must have chosen these defaults for some reason, I don't think it's a big deal either so let's merge this as it is? @LesnyRumcajs

**Pull request merged**